Session-4/CurrencyExchangeApp/CurrencyConverter/CurrencyConverter.cs

```diff
@@ -0,0 +1,80 @@
+﻿using Newtonsoft.Json;
+using System;
+using System.Configuration;
+using RestSharp;
+using System.Collections.Generic;
+using System.IO;
+
+
+namespace CurrencyExchange
+{
+    public class CurrencyConverter
+    {
+        public static string filePath = (string)AppDomain.CurrentDomain.GetData("FilePath");
+        private static ApiResponse GetApidata()
+        {
+            try {
+                RestClient client = new RestClient("https://api.apilayer.com/currency_data/live?source=INR&currencies=");
+                RestRequest request = new RestRequest();
+                request.AddHeader("apikey", "NKJlyNE9umLraoNvP87N7W9v6NNePbI8");
+                ApiResponse apiResponse = JsonConvert.DeserializeObject<ApiResponse>(client.Get(request).Content);
+                apiResponse.quotes.Add("timestamp", apiResponse.timestamp);
+                return apiResponse;
+            }
+            catch(Exception e) {
+                throw e.InnerException;
+            }
+                
+        }
+
+
+        public static double Convert(string fromCurrency, string toCurrency, double Currencyvalue)
+        {
+            
+            if (toCurrency == fromCurrency)
+                return 1.0;
+            string key = fromCurrency + toCurrency;
+            if (File.Exists(filePath))
```

Comment:
Instead of reading date from inside the file, include the date in file path iteself.

So if file exists with today's date > use it, otherwise get data from api and save that in today's date file.